sycl/test-e2e/Graph/Inputs/work_group_dynamic_memory.cpp

```diff
@@ -0,0 +1,65 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+//
+
+// Test work_group_dynamic extension with graph.
+
+#include <sycl/detail/core.hpp>
+#include <sycl/ext/oneapi/work_group_scratch_memory.hpp>
+#include "../graph_common.hpp"
+
+#include <vector>
+
+constexpr size_t WgSize = 32;
+constexpr size_t WgCount = 4;
+constexpr size_t RepeatWG = 16;
+constexpr size_t ElemPerWG = WgSize * RepeatWG;
+constexpr size_t Size = WgSize * WgCount * RepeatWG;
+
+using namespace sycl;
+
+namespace exp_ext = sycl::ext::oneapi::experimental;
+
+int main() {
+  queue Q;
+  exp_ext::command_graph Graph{Queue.get_context(), Queue.get_device()};
```

Comment:
Need to create this with the `exp_ext::property::graph::assume_buffer_outlives_graph{}` property to use buffers with Graphs